File: 2024-2/sesiones-de-clase/sesion-de-clase-15/Alumno.cpp
/* 
 * File:   Alumno.cpp
 * Author: erichuiza
 * 
 * Created on October 11, 2024, 4:45 PM
 */

#include "Alumno.h"

Alumno::Alumno() : Persona() {
    codigo = {};
    ciclo = {};
}

Alumno::Alumno(const Alumno& orig) : Persona(orig) {
}

Alumno::~Alumno() {
}

File: 2025-1/sesion-de-clase-22/src/Alumno.java
/**
 *
 * @author erichuiza
 */
public class Alumno extends Persona {
    private int ciclo;
    private int numCreditos;

    public Alumno() {
        super();
    }
    
    public Alumno(int dni, String nombre, int edad, 
            int ciclo, int numCreditos) {
        super(dni, nombre, edad);
        this.ciclo = ciclo;
        this.numCreditos = numCreditos;
    }
    
    public int getCiclo() {
        return ciclo;
    }

    public void setCiclo(int ciclo) {
        this.ciclo = ciclo;
    }

    public int getNumCreditos() {
        return numCreditos;
    }

    public void setNumCreditos(int numCreditos) {
        this.numCreditos = numCreditos;
    }

    @Override
    public void imprimir() {
        super.imprimir();
        System.out.println("Ciclo: " + ciclo);
        System.out.println("Num Créditos: " + numCreditos);
    }

    @Override
    public String toString() {
        String str = super.toString();
        str += "Ciclo: " + ciclo + "\n";
        str += "Num Creditos" + numCreditos  + "\n";
        
        return str;
    }
    
    
    
}
